File: backend/app/schemas/reports.py
from datetime import date
from decimal import Decimal

from pydantic import BaseModel


class ReportRequest(BaseModel):
    """Query params for the parameterized date-range report (CLAUDE.md §30
    M18) — one dispatcher, not one endpoint per cadence."""

    start_date: date
    end_date: date


class ReportResponse(BaseModel):
    start_date: date
    end_date: date
    total_disbursed: Decimal
    total_collected: Decimal
    total_expenses: Decimal
    net: Decimal
    par_percentage: Decimal
    active_loans: int

File: frontend/src/schemas/reports.ts
// Mirrors backend app/schemas/reports.py
export interface ReportResponse {
  start_date: string;
  end_date: string;
  total_disbursed: string;
  total_collected: string;
  total_expenses: string;
  net: string;
  par_percentage: string;
  active_loans: number;
}
